src/TMap.cpp

```diff
@@ -1953,24 +2063,12 @@ int TMap::createMapLabel(int area, QString text, float x, float y, float z, QCol
     QSizeF s = QSizeF(label.size.width() / zoom, label.size.height() / zoom);
     label.size = s;
     label.clickSize = s;
-    if (!mpRoomDB->getArea(area)) {
-        return -1;
-    }
 
-    int label_id;
+    int label_id = -1;
+    do {
+    } while (pA->mMapLabels.contains(++label_id));
```

Comment:
`createMapLabelID()` seems like a safer function for it, let's keep it.

Reply:
Well it is going to be much smaller than before - a simplistic reproduction would go from:
```cpp
int TMap::createMapLabelID(int area)
{
    if (mapLabels.contains(area)) {
        const QList<int> idList = mapLabels.value(area).keys();
        int id = 0;
        // protect against integer overflow
        while (id >= 0) {
            if (!idList.contains(id)) {
                return id;
            }
            id++;
        }
    }
    return -1;
}
```
to:
```
int TArea::createMapLabelID()
{
    int id = 0;
    // protect against integer overflow
    while (id >= 0) {
        if (!mapLabels.contains(id)) {
            return id;
        }
        id++;
    }
    return -1;
}
```
but the whole ideal of protecting against integer overflow seems like "*deckchair re-arrangement on H.M.S.S. Titanic*" - the application will have crashed from resource exhaustion long before we have enough map labels in an area for an `int` to overflow, IMHO.

Reply:
Prevent the Titanic one bit at a time